core/src/main/java/org/apache/iceberg/StaticTable.java

```diff
@@ -0,0 +1,262 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg;
+
+import java.io.Serializable;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import org.apache.iceberg.encryption.EncryptionManager;
+import org.apache.iceberg.io.FileIO;
+import org.apache.iceberg.io.LocationProvider;
+import org.apache.iceberg.relocated.com.google.common.collect.Maps;
+
+// TODO: any ideas on naming?
+public class StaticTable implements Table, Serializable {
```

Comment:
I did not want another Table implementation but Kryo will ignore the Java serialization lifecycle. That's why the proxy pattern we currently have won't work.

Reply:
Can we use this table implementation for the other case? It would be nice to have the proxy produce this class on `readResolve`. That way we only have one implementation. Spark would need to convert explicitly, but Java serialization would just work in other circumstances.

Reply:
Missed this comment. Yeah, this idea occurred to me too.

Reply:
One of the main goals of the `BaseTable` serialization was to create a way to start scans on the deserialized Table. You can see here the example:
https://github.com/apache/iceberg/blob/23735d1d99abf0207543ff5d9dcb63ae4fe4ec02/mr/src/main/java/org/apache/iceberg/mr/mapreduce/IcebergInputFormat.java#L97-L104

Maybe we can have Lazy loading of the metadata when needed and that way we can merge the two approach.

Reply:
How many times will we read the metadata file in this case? Just once, right, @pvary?

Reply:
@aokolnychyi, it would load metadata each time a scan is built. So if you called `newScan` on executors it would result in a lot of calls. But we don't expect that to happen so I think it should be okay.

Reply:
Yeah, we are fine in Spark. I just want to make sure that does not happen in Hive too.

Reply:
I replaced the proxies.

Reply:
I am resolving this thread as the current approach covers queries.